sql/analyzer/inserts.go

```diff
@@ -106,42 +107,43 @@ func existsNonZeroValueCount(values sql.Node) bool {
 	return false
 }
 
+func findColIdx(colName string, colNames []string) int {
+	for i, name := range colNames {
+		if strings.EqualFold(name, colName) {
+			return i
+		}
+	}
+	return -1
+}
+
 // wrapRowSource returns a projection that wraps the original row source so that its schema matches the full schema of
 // the underlying table, in the same order. Also returns a boolean value that indicates whether this row source will
 // result in an automatically generated value for an auto_increment column.
-func wrapRowSource(ctx *sql.Context, insertSource sql.Node, destTbl sql.Table, schema sql.Schema, columnNames []string) (sql.Node, bool, error) {
+func wrapRowSource(ctx *sql.Context, insertSource sql.Node, destTbl sql.Table, schema sql.Schema, columnNames []string) (sql.Node, int, error) {
 	projExprs := make([]sql.Expression, len(schema))
-	autoAutoIncrement := false
-
-	for i, f := range schema {
-		columnExplicitlySpecified := false
-		for j, col := range columnNames {
-			if strings.EqualFold(f.Name, col) {
-				projExprs[i] = expression.NewGetField(j, f.Type, f.Name, f.Nullable)
-				columnExplicitlySpecified = true
-				break
-			}
-		}
+	autoAutoIncrement := -1
```

Comment:
I know you didn't create the name `autoAutoIncrement`, but while you're refactoring, it seems like a good target to clean up. Something like `generatedAutoIncValueIdx` or `generatedAutoIncrementIdx` seems a bit easier to understand than `autoAuto<anything>`. Or... sticking with `UnspecifiedAutoIncrementIdx` and using that consistently could be nice, too.